src/VisualStudio/Core/Def/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -56,31 +57,38 @@ internal sealed partial class MiscellaneousFilesWorkspace : Workspace, IRunningD
 
         private readonly ForegroundThreadAffinitizedObject _foregroundThreadAffinitization;
 
+        private IVsTextManager _textManager;
+        private RunningDocumentTableEventTracker _runningDocumentTableEventTracker;
+
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public MiscellaneousFilesWorkspace(
             IThreadingContext threadingContext,
             IVsEditorAdaptersFactoryService editorAdaptersFactoryService,
             IMetadataAsSourceFileService fileTrackingMetadataAsSourceService,
-            VisualStudioWorkspace visualStudioWorkspace,
-            SVsServiceProvider serviceProvider)
+            VisualStudioWorkspace visualStudioWorkspace)
             : base(visualStudioWorkspace.Services.HostServices, WorkspaceKind.MiscellaneousFiles)
         {
             _foregroundThreadAffinitization = new ForegroundThreadAffinitizedObject(threadingContext, assertIsForeground: false);
 
+            _threadingContext = threadingContext;
+            _editorAdaptersFactoryService = editorAdaptersFactoryService;
             _fileTrackingMetadataAsSourceService = fileTrackingMetadataAsSourceService;
-            _lazyTextManager = new Lazy<IVsTextManager>(() =>
-            {
-                _foregroundThreadAffinitization.AssertIsForeground();
-                return (IVsTextManager)serviceProvider.GetService(typeof(SVsTextManager));
-            });
-
-            var runningDocumentTable = (IVsRunningDocumentTable)serviceProvider.GetService(typeof(SVsRunningDocumentTable));
-            _runningDocumentTableEventTracker = new RunningDocumentTableEventTracker(threadingContext, editorAdaptersFactoryService, runningDocumentTable, this);
 
             _metadataReferences = ImmutableArray.CreateRange(CreateMetadataReferences());
         }
 
+        public async Task InitializeAsync(IAsyncServiceProvider serviceProvider, CancellationToken cancellationToken)
+        {
+            _textManager = await serviceProvider.GetServiceAsync<SVsTextManager, IVsTextManager>(_threadingContext.JoinableTaskFactory).ConfigureAwait(false);
+            var runningDocumentTable = await serviceProvider.GetServiceAsync<SVsRunningDocumentTable, IVsRunningDocumentTable>(_threadingContext.JoinableTaskFactory).ConfigureAwait(false);
+
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
This can be removed, the running doc table advise/unadvise is explicitly free threaded now in VS. (See the comment in RunningDocumentTableEventTracker)